Fixed Bugs with Pause Menu and Dialogue Input

getmikyled opened the pull request:
BUGS FIXED:

- Cursor lock mode not returning to right state when pausing while dialogue is playing
- Fixed being able to advance dialogue while game is paused

getmikyled committed:
Fixed Bugs with Pause Menu and Dialogue Input

TheStress commented on Assets/Scripts/UI/PauseMenuUI.cs:
```diff
@@ -25,23 +27,30 @@ private void Update()
         }
     }
-    public void Resume()
+    public void Resume(CursorLockMode cursorLockMode)
```
Adding a parameter to this broke the pause menu prefab (the resume button)

getmikyled replied:
Fixed! I could not see anything in the pause menu UI prefab was testing it, I had assumed that it was not implemented yet.

TheStress commented on Assets/Scripts/UI/PauseMenuUI.cs:
```diff
@@ -11,6 +11,8 @@ public class PauseMenuUI : MonoBehaviour
     private bool gameIsPaused = false;
     public KeyCode pauseKey;
 
+    private CursorLockMode prevCursorLockMode;
```
you also need to keep track of the previous cursor visibility

getmikyled replied:
Fixed!

getmikyled committed:
Fixed Pause Menu Bugs

getmikyled committed:
Merge branch 'develop' into pausingInDialogueBugFix

TheStress committed:
fixed

- it looks like  assigning the visibility to false also assigns the lockState to locked also

- So I just swapped the saving to before the assigning

TheStress commented on Assets/Scripts/UI/PauseMenuUI.cs:
```diff
@@ -50,7 +60,12 @@ public void Pause()
         playerInput.DisableAbilityInput();
+
+        prevCursorVisibility = Cursor.visible;
+        prevCursorLockMode = Cursor.lockState;
```
While testing this I found that the issue still exists so I swapped reading the previous cursor state and the assignments of the new cursor state.

getmikyled replied:
Ah sorry it did not work. I did not see any issues on my end. Weird fix though, I don't see any issues with the logic.

TheStress reviewed (approved):
Looks good
